### PR TITLE
fix: UB during cmd squashing reply size calculation

### PR DESCRIPTION
**Pull request by BorysTheDev**
address comments from https://github.com/dragonflydb/dragonfly/pull/4147

### CODE REVIEW

File: src/server/multi_command_squasher.cc [hunk 1]
```diff
@@ -273,8 +273,8 @@ bool MultiCommandSquasher::ExecuteSquashed(facade::RedisReplyBuilder* rb) {
 
     aborted |= error_abort_ && CapturingReplyBuilder::TryExtractError(replies.back());
 
-    CapturingReplyBuilder::Apply(std::move(replies.back()), rb);
     current_reply_size_.fetch_sub(Size(replies.back()), std::memory_order_relaxed);
+    CapturingReplyBuilder::Apply(std::move(replies.back()), rb);
```
Comment (kostasrim) [inline src/server/multi_command_squasher.cc:277]: Only question here is why do you think `std::move()` does not move here? (based your comments on previous PR). To me, `Apply` accepts an `rvalue reference` of type Payload (a variant), `replies.back()` returns an lvalue ref which is properly casted "moved" to an rvalue. So we do actually move it ?

Comment (BorysTheDev) [reply src/server/multi_command_squasher.cc:277]: reference cast does nothing, it is still a reference. CapturingReplyBuilder::Apply() doesn't actually modify the moved object, so there is no difference if have std::move or not

**Comment by adiholden** [comment]
@BorysTheDev 
please add the stats to PrintPrometheusMetrics
take pipeline_cmd_cache_bytes for example

File: src/server/multi_command_squasher.cc (hunk 1 above)
Comment (kostasrim) [reply src/server/multi_command_squasher.cc:277]: Discussed this on a call. Borys is right, we don't move here, it's a reference

**Review by adiholden — APPROVED** [review]

**Pull request merged**